### PR TITLE
Pull Dev Branch to Master: Enable/Disable Auditing Feature with Docs

### PR DESCRIPTION
**Pull request by bongiovimatthew-microsoft**

### CODE REVIEW

File: AdfsEventsModule.psm1 [hunk 1]
```diff
@@ -4,6 +4,130 @@
...
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6001'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6002'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7600'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7601'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'9200'{Import-Module ADFS -ErrorAction SilentlyContinue}
+		'9600'{Import-Module ADFS -ErrorAction SilentlyContinue}
+		'14393'{Import-Module ADFS -ErrorAction SilentlyContinue}
```
Comment (tquerec) [inline AdfsEventsModule.psm1]: Should we add a default case to future proof this?

File: AdfsEventsModule.psm1 [hunk 2]
```diff
@@ -4,6 +4,130 @@
...
+	}
+
+    $SyncProps = Get-ADFSSyncProperties
+    if ( $SyncProps.Role -ne 'SecondaryComputer' ) 
+    {
+        Set-ADFSProperties -LogLevel  @( "Warnings", "Errors", "Information" )
+    }
+	auditpol.exe /set /subcategory:"Application Generated" /failure:disable /success:disable
+	
+	Write-host "ADFS auditing is now disabled."
```
Comment (tquerec) [inline AdfsEventsModule.psm1]: Writing to the host will mess with chaining to other scripts.  Use write-verbose instead

File: AdfsEventsModule.psm1 [hunk 3]
```diff
@@ -4,6 +4,130 @@
...
+	$OSVersion = gwmi win32_operatingsystem
+	[int]$BuildNumber = $OSVersion.BuildNumber 
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6001'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'6002'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7600'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'7601'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
+		'9200'{Import-Module ADFS -ErrorAction SilentlyContinue}
```
Comment (tquerec) [inline AdfsEventsModule.psm1]: This block is the same as above.  Factor into a separate function?

File: AdfsEventsModule.psm1 [hunk 4]
```diff
@@ -4,6 +4,130 @@
...
+	}
+
+	#Disable security auditing from ADFS
+	$cs = get-wmiobject -class win32_computersystem -ComputerName "localhost"
+	$DomainRole = $cs.domainrole
+	$OSVersion = gwmi win32_operatingsystem
+	[int]$BuildNumber = $OSVersion.BuildNumber 
+	switch ($OSVersion.Buildnumber)
+	{
+		'6000'{Add-PsSnapin Microsoft.Adfs.Powershell -ErrorAction SilentlyContinue}
```
Comment (tquerec) [inline AdfsEventsModule.psm1]: Are you sure you want to silentlycontinue if the module can't be loaded?  Isn't that a fatal error for this script?

File: AdfsEventsModule.psm1 (hunk 2 above)
Comment (bongiovimatthew-microsoft) [reply AdfsEventsModule.psm1]: Good catch. Fixing in next iteration

File: AdfsEventsModule.psm1 (hunk 3 above)
Comment (bongiovimatthew-microsoft) [reply AdfsEventsModule.psm1]: Agreed, moving to its own function

File: AdfsEventsModule.psm1 (hunk 1 above)
Comment (bongiovimatthew-microsoft) [reply AdfsEventsModule.psm1]: Agreed. Changing this to a single if for build num less than or equal to 7601

File: AdfsEventsModule.psm1 (hunk 4 above)
Comment (bongiovimatthew-microsoft) [reply AdfsEventsModule.psm1]: If we fail to load the PowerShell, then the only thing we can't enable is the log level in Adfs Properties. Do you think there are error scenarios where the module might fail to load when we call it, but it is already loaded successfully? In that case, failing silently will lead to a success

**Pull request merged**